### PR TITLE
[docs] Improve documentation about batching

### PR DESCRIPTION
**Pull request by gmethvin**
I noticed the documentation around batching didn't clarify some important details about how batching works in Pulsar, and particularly how it interacts with acknowledgements and redeliveries. This is my attempt to provide some clarity.

### CODE REVIEW

File: site2/docs/concepts-messaging.md [hunk 1]
```diff
@@ -49,7 +49,14 @@ Messages published by producers can be compressed during transportation in order
...
 ### Batching
 
-If batching is enabled, the producer will accumulate and send a batch of messages in a single request. Batching size is defined by the maximum number of messages and maximum publish latency.
+If batching is enabled, the producer will accumulate and send a batch of messages in a single request. Batch size is defined by the maximum number of messages and maximum publish latency.
+
+Batches are tracked and stored by Pulsar as batches rather than as individual messages. Under the hood the consumer unbundles these batches into individual messages. Since the messages are stored as batches, the backlog size will also represent the total number of batches rather than the total number of messages.
+
+Scheduled messages (using `deliverAt` or `deliverAfter`) are always sent as individual messages even when batching is enabled.
+
+> Note
```
Comment (Lanayx) [inline site2/docs/concepts-messaging.md]: I think just noting is not enough, it should be pointed out that this is a buggy behavior and should be fixed in future releases.

File: site2/docs/concepts-messaging.md [hunk 2]
```diff
@@ -49,7 +49,14 @@ Messages published by producers can be compressed during transportation in order
...
+> Since batches are tracked as single units, a batch will only be considered acknowledged when all its messages are acknowledged by the consumer. This means unexpected failures, negative acknowledgements, and acknowledgement timeouts can result in redelivery of all messages in the batch, even if some of the messages have already been acknowledged.
```
Comment (Lanayx) [inline site2/docs/concepts-messaging.md]: I think just noting is not enough, it should be pointed out that this is a buggy behavior and should be fixed in future releases.

Comment (cdbartholomew) [reply site2/docs/concepts-messaging.md]: See  https://github.com/apache/pulsar/pull/5990 for a fix for this from @zzzming.

Comment (gmethvin) [reply site2/docs/concepts-messaging.md]: That fix is only on the client side, so batches may still be fully redelivered if the consumer crashes or restarts, though.

**Comment by gmethvin** [comment]
anyone know why is CI not running for this PR?

File: site2/docs/concepts-messaging.md (hunk 2 above)
Comment (cdbartholomew) [reply site2/docs/concepts-messaging.md]: True. That fix improves the situation, but doesn't resolve it entirely.

**Comment by sijie** [comment]
it seems that closing the PR doesn't trigger CI. @tuteng can you help check why the CI doesn't run for this pull request?

File: site2/docs/concepts-messaging.md (hunk 2 above)
Comment (gmethvin) [reply site2/docs/concepts-messaging.md]: If we agree it's definitely a bug, then let's get the fix and this documentation update in the next release, and I'll remove the part about negative acks.

Comment (sijie) [reply site2/docs/concepts-messaging.md]: Agreed. This is a bug and fixes at the client side are not enough. Just fixing this at the client side is actually making the behavior much more confused. We need to get this documentation merged first and then fix the behavior fundamentally in the next release.

**Comment by sijie** [comment]
since the master is back to normal, trigger CI test via " retest this please "

**Comment by jiazhai** [comment]
run java8 tests

**Comment by sijie** [comment]
retest this please

**Comment by sijie** [comment]
run java8 tests
run cpp tests

**Pull request merged**